source/common/upstream/upstream_impl.h

```diff
@@ -101,6 +102,56 @@ class UpstreamLocalAddressSelectorImpl : public UpstreamLocalAddressSelector {
   std::vector<UpstreamLocalAddress> upstream_local_addresses_;
 };
 
+/**
+ * Class for LBPolicies
+ * Uses a absl::variant to store pointers for the LBPolicy
+ */
+class LBPolicyConfig {
+public:
+  LBPolicyConfig(const envoy::config::cluster::v3::Cluster& config);
+
+  template <typename T> OptRef<const T> getConfig() const {
+    if (absl::holds_alternative<std::unique_ptr<const T>>(lbPolicy_)) {
+      if (absl::get<std::unique_ptr<const T>>(lbPolicy_) != nullptr)
+        return *(absl::get<std::unique_ptr<const T>>(lbPolicy_));
```

Comment:
nit:
```suggestion
      if (auto config = absl::get<std::unique_ptr<const T>>(lbPolicy_); config != nullptr)
        return *config;
```

Reply:
I'm not sure if this would work. Unless I misunderstood, I thought you couldn't copy a unique ptr?

Reply:
I see this pattern:
```
if (func() != nullptr) {
  return *func();
}
```
and I suggested replacing it with:
```
if (auto var = func(); var != nullptr) {
  return *var;
}
```

Let's delay this discussion until we see if `get_if` should be used instead.